### DISCUSSION TITLE
Allure JavaScript integrations 2.7.0: Playwright improvements

### DISCUSSION THREAD

**Discussion by baev** [allure-framework/allure2#2103] [Announcements]
# 🎉 Exciting Allure Playwright Update! 🚀

We are thrilled to announce the release of Allure Playwright 2.7.0, packed with powerful enhancements and improvements that will elevate your testing experience. 

1. 📎 **Step Attachments with `allure.attachment` API** 
You asked, and we delivered! Say hello to a long-awaited feature - support for step attachments. We're introducing the all-new `allure.attachment` API. With this addition, you can add attachments to individual test steps, providing more context and making your test reports even more insightful. While we still support the use of `test.info().attach`, we highly recommend replacing it with `allure.attachment` to leverage this powerful feature.
2. 🚶‍♂️ **`allure.step` - Your Step Companion** 
Introducing the brand-new `allure.step method`! Think of it as an alias for test.step designed for convenience. 
3. 📋 **`allure.logStep` - Log Steps for Clarity**
Need to log additional details to your reports? We've got you covered with the introduction of allure.logStep. These are steps without a body, ideal for documenting crucial points in your test flow. Now, your reports will be more organized and informative than ever.
4. ⏳ **Await Allure API Methods** 
We've made a crucial update to ensure the completeness and accuracy of your test metadata. All Allure API methods must now be awaited. Don't miss out on essential metadata — add the await keyword to your API calls and enjoy more reliable and detailed reports.
5.  🐞 **`detail` and `suiteTitle` options are now true by default.**
Before the change, the behaviour was unclear, making defaults to true if no reporter options are specified but changing defaults to false if at least one option is present. The fix is to change the behaviour to make both of the options always true by default.
6. 💚 **Allure TestOps Users - Improved `testCaseId` Generation** 
For our Allure TestOps users, we've revamped the testCaseId generation algorithm. This change allows for better support of tests with the same names within a single file if you don't use allure.id, duplicate test cases may be created. Utilize allure.id to keep your test cases neatly organized.

We understand these changes may require some adjustment, but they are designed to elevate your testing capabilities.

As always, we value your feedback and are here to support you during this transition. If you have any questions or need assistance, please don't hesitate to contact our [community forum](https://github.com/orgs/allure-framework/discussions) or support channels.

As usual, the full release notes are available [here](https://github.com/allure-framework/allure-js/releases/tag/v2.7.0).

Happy testing!

Best regards,
Dmitry

**Comment by amaan-02** [comment]
i want to integrate with GitHub action with ci cd for artifacts upload allure report in playwright project of workflow. NodeJS project
